### PR TITLE
Add -nt ${GALAXY_SLOTS:-4} to mdrun

### PR DESCRIPTION
**Pull request by simonbray**
Stop GROMACS job trying to use every core on the node they are assigned to.

### CODE REVIEW

File: tools/gromacs/minim.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
         ln -s '$top_input' ./top_input.top &&
   
         gmx grompp -f ./minim.mdp -c ./solv_ions.gro -r ./solv_ions.gro -p ./top_input.top -o em.tpr &>> verbose.txt &&
-        gmx mdrun -deffnm em &>> verbose.txt
+        gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm em &>> verbose.txt
```
Comment (simonbray) [inline tools/gromacs/minim.xml]: @bgruening, don't really know whether 4 is a good choice here.

Comment (bgruening) [reply tools/gromacs/minim.xml]: I think this is ok!

Comment (simonbray) [reply tools/gromacs/minim.xml]: can we add this to @tsenapathi's bash script as well? e.g. here: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/blob/master/tools/free_energy/alchemical_run/gmx_fep.sh#L50

Comment (bgruening) [reply tools/gromacs/minim.xml]: We should! GALAXY_SLOTS is a real ENV variable so it should also work in the bash script.

Comment (simonbray) [reply tools/gromacs/minim.xml]: cool

**Comment by bgruening** [comment]
Mh, Fatal error: Exit code 139 ()

File: tools/free_energy/alchemical_run/gmx_fep.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ do
 
     gmx grompp -f $MDP/em_steep_$LAMBDA.mdp -c $FREE_ENERGY/morph.gro -p $FREE_ENERGY/morph.top -o min$LAMBDA.tpr
 
-    gmx mdrun -deffnm min$LAMBDA
+    gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm min$LAMBDA
```
Comment (simonbray) [inline tools/free_energy/alchemical_run/gmx_fep.sh]:
```suggestion
    gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm min$LAMBDA
```

File: tools/free_energy/alchemical_run/gmx_fep.sh [hunk 3]
```diff
@@ -62,7 +62,7 @@ do
 
     gmx grompp -f $MDP/nvt_$LAMBDA.mdp -c ../EM/min$LAMBDA.gro -p $FREE_ENERGY/morph.top -o nvt$LAMBDA.tpr
 
-    gmx mdrun -deffnm nvt$LAMBDA
+    gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm nvt$LAMBDA
```
Comment (simonbray) [inline tools/free_energy/alchemical_run/gmx_fep.sh]:
```suggestion
    gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm min$LAMBDA
```

File: tools/free_energy/alchemical_run/gmx_fep.sh [hunk 4]
```diff
@@ -78,7 +78,7 @@ do
 
     gmx grompp -f $MDP/npt_$LAMBDA.mdp -c ../NVT/nvt$LAMBDA.gro -p $FREE_ENERGY/morph.top -t ../NVT/nvt$LAMBDA.cpt -o npt$LAMBDA.tpr
 
-    gmx mdrun -deffnm npt$LAMBDA
+    gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm npt$LAMBDA
```
Comment (simonbray) [inline tools/free_energy/alchemical_run/gmx_fep.sh]:
```suggestion
    gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm min$LAMBDA
```

File: tools/free_energy/alchemical_run/gmx_fep.sh [hunk 5]
```diff
@@ -94,7 +94,7 @@ do
 
     gmx grompp -f $MDP/md_$LAMBDA.mdp -c ../NPT/npt$LAMBDA.gro -p $FREE_ENERGY/morph.top -t ../NPT/npt$LAMBDA.cpt -o md$LAMBDA.tpr
 
-    gmx mdrun -deffnm md$LAMBDA
+    gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm md$LAMBDA
```
Comment (simonbray) [inline tools/free_energy/alchemical_run/gmx_fep.sh]:
```suggestion
    gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm min$LAMBDA
```

File: tools/gromacs/minim.xml (hunk 1 above)
Comment (simonbray) [inline tools/gromacs/minim.xml]:
```suggestion
        gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm em &>> verbose.txt
```

File: tools/gromacs/sim.xml [hunk 6]
```diff
@@ -44,7 +44,7 @@
             -o outp.tpr &>> verbose.txt &&
 
 
-        gmx mdrun -deffnm outp &>> verbose.txt
+        gmx mdrun -nt ${GALAXY_SLOTS:-4} -deffnm outp &>> verbose.txt
```
Comment (simonbray) [inline tools/gromacs/sim.xml]:
```suggestion
        gmx mdrun -nt "\${GALAXY_SLOTS:-4}" -deffnm outp &>> verbose.txt
```

**Comment by simonbray** [comment]
Thanks @bgruening - can you also update the tools on usegalaxy.eu?

**Comment by bgruening** [comment]
I can't it failed on master ... and I do not know why :(

**Comment by bgruening** [comment]
tools are updated ...

**Pull request merged**